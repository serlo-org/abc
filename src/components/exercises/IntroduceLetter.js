import React from 'react';
import { View, Dimensions } from 'react-native';

import { GREEN } from '../../styles/colors';
import { RoundImageWithBorder, RoundText } from '../Components';

const ScaleFourImages = ({ displaySize, images, letter }) => {
  const scale = displaySize / 360;
  return (
    <View
      style={{
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center'
      }}
    >
      <RoundText
        style={{
          marginLeft: 35 * scale
        }}
        text={letter}
        textStyle={{
          fontSize: 50 * scale
        }}
        size={130 * scale}
      />
      <View
        style={{
          flexDirection: 'row',
          alignItems: 'center',
          marginTop: -27 * scale
        }}
      >
        <RoundImageWithBorder
          image={images[0]}
          size={130 * scale}
          white
          style={{ margin: 0 }}
        />
        <RoundImageWithBorder
          image={images[1]}
          size={110 * scale}
          white
          style={{ margin: 0 }}
        />
      </View>
      <RoundImageWithBorder
        style={{ marginTop: -25 * scale, marginLeft: 50 * scale }}
        image={images[2]}
        size={130 * scale}
        white
      />
    </View>
  );
};
const IntroduceLetter = props => {
  const { width } = Dimensions.get('window');
  return (
    <View
      style={{
        flex: 1,
        backgroundColor: GREEN,
        alignItems: 'center',
        justifyContent: 'center'
      }}
    >
      <ScaleFourImages displaySize={width} {...props} />
    </View>
  );
};

export default IntroduceLetter;
