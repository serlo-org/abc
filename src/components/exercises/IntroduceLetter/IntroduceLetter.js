import React from 'react';
import { View, Dimensions } from 'react-native';

import { GREEN } from '../../../styles/colors';
import RoundImageWithBorder from '../../common/RoundImageWithBorder';
import RoundText from '../../common/RoundText';
import { PortraitScreenOrientation } from '../../helpers/screen-orientation';

const ScaleFourImages = ({ displaySize, letter, words }) => {
  const scale = displaySize / 360;
  return (
    <View
      style={{
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center'
      }}
    >
      <RoundText
        style={{
          marginLeft: 35 * scale
        }}
        text={letter}
        textStyle={{
          fontSize: 50 * scale
        }}
        size={130 * scale}
      />
      <View
        style={{
          flexDirection: 'row',
          alignItems: 'center',
          marginTop: -27 * scale
        }}
      >
        <RoundImageWithBorder
          image={words[0].getImage()}
          size={130 * scale}
          white
          style={{ margin: 0 }}
        />
        <RoundImageWithBorder
          image={words[1].getImage()}
          size={110 * scale}
          white
          style={{ margin: 0 }}
        />
      </View>
      <RoundImageWithBorder
        style={{ marginTop: -25 * scale, marginLeft: 50 * scale }}
        image={words[2].getImage()}
        size={130 * scale}
        white
      />
    </View>
  );
};

const IntroduceLetter = props => {
  const { width } = Dimensions.get('window');
  return (
    <PortraitScreenOrientation>
      <View
        style={{
          flex: 1,
          backgroundColor: GREEN,
          alignItems: 'center',
          justifyContent: 'center'
        }}
      >
        <ScaleFourImages displaySize={width} {...props} />
      </View>
    </PortraitScreenOrientation>
  );
};

export default IntroduceLetter;
