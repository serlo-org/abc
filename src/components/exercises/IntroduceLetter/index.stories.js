import { storiesOf } from '@storybook/react-native';

import * as IntroduceLetter from '.';
import { createStoriesFromFixtures } from '../helpers';

const story = storiesOf('exercises/IntroduceLetter', module);

createStoriesFromFixtures(story, IntroduceLetter);
