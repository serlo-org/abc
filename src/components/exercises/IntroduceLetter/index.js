import { getWordObject } from '../../../helpers/words';
import { InfoScreen } from '../../../exercises';
import Component from './IntroduceLetter';

export const Exercise = InfoScreen.Exercise;
export const fixtures = InfoScreen.createFixtures([
  {
    name: 'A',
    props: {
      words: [
        getWordObject('ananas'),
        getWordObject('apfel'),
        getWordObject('affe')
      ],
      letter: 'Aa'
    }
  }
]);
export { Component };
