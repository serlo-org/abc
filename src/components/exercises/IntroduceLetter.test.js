import React from 'react';

import IntroduceLetter from './IntroduceLetter';

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const tree = renderer.create(
    <IntroduceLetter
      images={[
        require('../assets/images/ananas.jpg'),
        require('../assets/images/apfel.jpg'),
        require('../assets/images/affe.jpg')
      ]}
      letter="Aa"
    />
  );
  expect(tree).toMatchSnapshot();
});
