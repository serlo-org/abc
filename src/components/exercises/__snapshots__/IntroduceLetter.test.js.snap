// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#b5ce4d",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 9999,
            "height": 135.41666666666669,
            "justifyContent": "center",
            "padding": 5,
            "width": 135.41666666666669,
          },
          Object {
            "marginLeft": 36.458333333333336,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontSize": 40,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            Object {
              "fontSize": 52.083333333333336,
            },
          ]
        }
      >
        Aa
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": -28.125000000000004,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0.2)",
              "borderRadius": 9999999,
              "margin": 13.541666666666668,
            },
            Object {
              "margin": 0,
            },
          ]
        }
      >
        <Image
          resizeMode="cover"
          source={1}
          style={
            Object {
              "borderRadius": 67.70833333333334,
              "height": 135.41666666666669,
              "margin": 13.541666666666668,
              "width": 135.41666666666669,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0.2)",
              "borderRadius": 9999999,
              "margin": 11.458333333333334,
            },
            Object {
              "margin": 0,
            },
          ]
        }
      >
        <Image
          resizeMode="cover"
          source={1}
          style={
            Object {
              "borderRadius": 57.29166666666667,
              "height": 114.58333333333334,
              "margin": 11.458333333333334,
              "width": 114.58333333333334,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.2)",
            "borderRadius": 9999999,
            "margin": 13.541666666666668,
          },
          Object {
            "marginLeft": 52.083333333333336,
            "marginTop": -26.041666666666668,
          },
        ]
      }
    >
      <Image
        resizeMode="cover"
        source={1}
        style={
          Object {
            "borderRadius": 67.70833333333334,
            "height": 135.41666666666669,
            "margin": 13.541666666666668,
            "width": 135.41666666666669,
          }
        }
      />
    </View>
  </View>
</View>
`;
